tests/app/dao/test_ft_billing_dao.py

```diff
@@ -230,7 +248,48 @@ def test_fetch_annual_billing_for_year_adds_data_for_today(notify_db_session):
     create_notification(template=template, status='delivered')
 
     assert db.session.query(FactBilling.bst_date).count() == 31
-    results = fetch_montly_billing_for_year(service_id=service.id,
-                                            year=2018)
+    results = fetch_monthly_billing_for_year(service_id=service.id,
+                                             year=2018)
     assert db.session.query(FactBilling.bst_date).count() == 32
     assert len(results) == 2
+
+
+def test_fetch_monthly_billing_for_year_return_financial_year(notify_db_session):
+    service = create_service()
+    sms_template = create_template(service=service, template_type="sms")
+    email_template = create_template(service=service, template_type="email")
+    letter_template = create_template(service=service, template_type="letter")
+    for month in range(1, 13):
+        mon = str(month).zfill(2)
+        days_in_month = {1: 32, 2: 30, 3: 32, 4: 31, 5: 32, 6: 31, 7: 32, 8: 32, 9: 31, 10: 32, 11: 31, 12: 32}
+        for day in range(1, days_in_month[month]):
+            d = str(day).zfill(2)
+            create_ft_billing(bst_date='2016-{}-{}'.format(mon, d),
+                              service=service,
+                              template=sms_template,
+                              notification_type='sms',
+                              rate=0.162)
+            create_ft_billing(bst_date='2016-{}-{}'.format(mon, d),
+                              service=service,
+                              template=email_template,
+                              notification_type='email',
+                              rate=0)
+            create_ft_billing(bst_date='2016-{}-{}'.format(mon, d),
+                              service=service,
+                              template=letter_template,
+                              notification_type='letter',
+                              rate=0.33)
+
+    results = fetch_monthly_billing_for_year(service.id, 2016)
+    # returns 3 rows, per month, returns financial year april to december
+    # Orders by Month
+
+    assert len(results) == 27
```

Comment:
3 rows per month should be 36?

Reply:
Ah - that's confusing, the data is only populated until December so there is 9 months in the test